tests/base/src/test/java/org/wildfly/security/http/digest/DigestAuthenticationMechanismTest.java

```diff
@@ -189,20 +299,46 @@ public void testSha512_256() throws Exception {
         Assert.assertEquals(UNAUTHORIZED, response.getStatusCode());
         Assert.assertEquals("Digest realm=\"api@example.org\", nonce=\"5TsQWLVdgBdmrQ0XsxbDODV+57QdFR34I9HAbC/RVvkK\", opaque=\"00000000000000000000000000000000\", algorithm=SHA-512-256, qop=auth", response.getAuthenticateHeader());
 
-        TestingHttpServerRequest request2 = new TestingHttpServerRequest(new String[] {
+        testStatefulDigest(new String[] {
                 "Digest username*=UTF-8''J%C3%A4s%C3%B8n%20Doe,\n" +
-                "       realm=\"api@example.org\",\n" +
-                "       uri=\"/doe.json\",\n" +
-                "       algorithm=SHA-512-256,\n" +
-                "       nonce=\"5TsQWLVdgBdmrQ0XsxbDODV+57QdFR34I9HAbC/RVvkK\",\n" +
-                "       nc=00000001,\n" +
-                "       cnonce=\"NTg6RKcb9boFIAS3KrFK9BGeh+iDa/sm6jUMp2wds69v\",\n" +
-                "       qop=auth,\n" +
-                "       response=\"3798d4131c277846293534c3edc11bd8a5e4cdcbff78b05db9d95eeb1cec68a5\",\n" +
-                "       opaque=\"00000000000000000000000000000000\",\n" +
-                "       userhash=false"
-        });
-        mechanism.evaluateRequest(request2);
-        Assert.assertEquals(Status.COMPLETE, request2.getResult());
+                        "       realm=\"api@example.org\",\n" +
+                        "       uri=\"/doe.json\",\n" +
+                        "       algorithm=SHA-512-256,\n" +
+                        "       nonce=\"5TsQWLVdgBdmrQ0XsxbDODV+57QdFR34I9HAbC/RVvkK\",\n" +
+                        "       nc=00000001,\n" +
+                        "       cnonce=\"NTg6RKcb9boFIAS3KrFK9BGeh+iDa/sm6jUMp2wds69v\",\n" +
+                        "       qop=auth,\n" +
+                        "       response=\"3798d4131c277846293534c3edc11bd8a5e4cdcbff78b05db9d95eeb1cec68a5\",\n" +
+                        "       opaque=\"00000000000000000000000000000000\",\n" +
+                        "       userhash=false"
+        },mechanism);
+
+        //test case insensitive
```

Comment:
Hi @keshav-redhat , thank you for the PR! Just a minor, but adding a single test for case insensitivity is sufficient. We do not have to add case insensitive test for each test method here